### ISSUE TITLE
[BUG] $name is an array with null on the ZugferdDocumentReader::getDocumentAdditionalReferencedDocument()

### ISSUE THREAD

**Issue by tommyseus** [horstoeko/zugferd#130]
Hello, I have a Problem in the `ZugferdDocumentReader::getDocumentAdditionalReferencedDocument()` function.

`$name` is an array with null as value `[null]`.

The Bug is new since [v1.0.62](https://github.com/horstoeko/zugferd/releases/tag/v1.0.62).

The problem is that `ReferencedDocumentType::getName()` returns a string and not TextType object. Not possible to access the `value` property of the TextType object.

Some tests are checking if the `$name` is an non-empty array, but not if the value in the array is a string.

https://github.com/horstoeko/zugferd/blob/35caab8f8b11a8c6e1ab3d168d30a4548e648214/src/ZugferdDocumentReader.php#L2454

This maybe fixes the issue:
```
$name = $this->getObjectHelper()->ensureArray($this->getInvoiceValueByPathFrom($addRefDoc, "getName", []));
```

**Comment by horstoeko** [comment]
Hi @tommyseus,

thank you for this issue. I will have a look on it... Stay tuned...

Kind regards

**Comment by horstoeko** [comment]
Hi @tommyseus,

I searched the history and found out that the schema definition (XSD) specified an incorrect cardinality for the name at the time. That's why the name is still an array (to avoid breaking changes). And somehow the type definition (in the YAML file) slipped through during the changeover to ZUGFeRD 2.3. I have made a fix and a new release. I hope this helps you. I hope to get some feedback from you.

Kind regards
